ddtrace/tracer/log_test.go

```diff
@@ -259,3 +259,17 @@ func setup(t *testing.T, customProp Propagator) string {
 	require.Len(t, tp.Logs(), 2)
 	return tp.Logs()[1]
 }
+
+func TestAgentURL(t *testing.T) {
+	assert := assert.New(t)
+	tp := new(log.RecordLogger)
+	tracer := newTracer(WithLogger(tp), WithUDS("var/run/datadog/apm.socket"))
+	defer tracer.Stop()
+	tp.Reset()
+	tp.Ignore("appsec: ", telemetry.LogPrefix)
+	logStartup(tracer)
+	for _, log := range tp.Logs() {
+		fmt.Printf("hello, %v\n\n", log)
+	}
+	assert.Regexp(`"agent_url":"unix://var/run/datadog/apm.socket"`, tp.Logs()[1])
+}
```

Comment:
Let's add a couple more test cases:

1. A test case for DD_TRACE_AGENT_URL = unix://var/run/datadog/apm.socket (in this case, you'd initialize `tracer` without the `WithUDS` option, e.g, `tracer := newTracer(WithLogger(tp))`
2. A test case for an invalid socket address
3. A test case for when both UDS and TCP configuration has been attempted (e.g,  DD_TRACE_AGENT_URL = unix://var/run/datadog/apm.socket + initializing the tracer with [WithAgentAddr](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1@v1.69.1/ddtrace/tracer#WithAgentAddr) opt pointed at http address e.g, localhost:8126, etc)